e2e/tests/plugins/timeConductor/timeConductor.e2e.spec.js

```diff
@@ -73,40 +73,119 @@ test.describe('Time counductor operations', () => {
 
-        // Click time offset button
-        await page.locator('.c-conductor__delta-button >> text=00:30:00').click();
+        // Set end time offset
+        await setOffset(page, endOffset, false);
 
-        // Input start time offset
-        await page.fill('.pr-time-controls__secs', '23');
+        // Verify time was updated on preceding time offset button
+        await expect(page.locator('.c-conductor__delta-button').nth(1)).toContainText('00:00:31');
+    });
 
-        // Click the check button
-        await page.locator('.icon-check').click();
+    /**
+     * Verify that start and end offsets are preserved when switching between
+     * fixed timespan and real-time mode.
+     */
+    test('preserve offsets when switching between fixed and real-time mode', async ({ page }) => {
+        const startOffset = {
+            secs: '23'
+        };
 
-        // Verify time was updated on time offset button
-        await expect(page.locator('.c-conductor__delta-button').first()).toContainText('00:30:23');
+        const endOffset = {
+            secs: '00'
+        };
 
-        // Click time offset set preceding now button
-        await page.locator('.c-conductor__delta-button >> text=00:00:30').click();
+        //Go to baseURL
+        await page.goto('/', { waitUntil: 'networkidle' });
 
-        // Input preceding time offset
-        await page.fill('.pr-time-controls__secs', '31');
+        // Switch to real-time mode
+        setTimeConductorMode(page, false);
 
-        // Click the check buttons
-        await page.locator('.icon-check').click();
+        // Set start time offset
+        await setOffset(page, startOffset, true);
 
-        // Verify time was updated on preceding time offset button
-        await expect(page.locator('.c-conductor__delta-button').nth(1)).toContainText('00:00:31');
+        // Set end time offset
+        await setOffset(page, endOffset, false);
+
+        // Switch to fixed timespan mode
+        setTimeConductorMode(page, true);
+
+        // Switch back to real-time mode
+        setTimeConductorMode(page, false);
+
+        // Verify updated start time offset persists after mode switch
+        await expect(page.locator('.c-conductor__delta-button').first()).toContainText('00:30:23');
+
+        // Verify updated end time offset persists after mode switch
+        await expect(page.locator('.c-conductor__delta-button').nth(1)).toContainText('00:00:00');
     });
 });
+
+/**
+ * @typedef {Object} OffsetValues
+ * @property {string | undefined} hours
+ * @property {string | undefined} mins
+ * @property {string | undefined} secs
+ */
+
+/**
+ * @param {import('@playwright/test').Page} page
+ * @param {OffsetValues} offset
+ * @param {boolean} isStartOffset
+ */
+async function setOffset(page, {hours, mins, secs}, isStartOffset = true) {
+    if (isStartOffset) {
+        await page.locator('.c-conductor__delta-button').first().click();
```

Comment:
This is a great implementation, but kinda illustrates the problem with our current class name locator strategy

This is where it would be great to update the locator of the object to be named something like `testid=condutor_start_button` so that we can remove some complexity in the test code. Without making those changes, this is a great strategy

Reply:
Totally agree. Since adding a `testid` attribute to the element won't impact anything else, I'd like to go ahead and do that here and refactor the tests.

Do you have any suggestions for `testid` nomenclature? In trying to keep all the details, I've come up with the rather verbose `conductor-realtime-start-offset-button` and `conductor-realtime-end-offset-button`.